app/api/groups/update_group_test.go

```diff
@@ -0,0 +1,137 @@
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			r, _ := http.NewRequest("POST", "/", strings.NewReader(tt.json))
+			_, err := validateUpdateGroupInput(r)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("validateUpdateGroupInput() error = %#v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func TestService_updateGroup_ErrorOnReadInTransaction(t *testing.T) {
+	assertUpdateGroupFailsOnDBErrorInTransaction(t, func(mock sqlmock.Sqlmock) {
+		mock.ExpectBegin()
+		mock.ExpectQuery(`SELECT .+ WHERE \(users\.ID = \?\)`).WithArgs(2).WillReturnError(errors.New("error"))
+		mock.ExpectQuery(regexp.QuoteMeta("SELECT groups.bFreeAccess FROM `groups` JOIN groups_ancestors ON groups_ancestors.idGroupChild = groups.ID WHERE (groups_ancestors.idGroupAncestor=?) AND (groups.ID = ?) LIMIT 1 FOR UPDATE")).
+			WithArgs(0, 1).WillReturnError(errors.New("some error"))
+		mock.ExpectRollback()
+	})
+}
+
+func TestService_updateGroup_ErrorOnRefusingSentGroupRequests(t *testing.T) {
+	assertUpdateGroupFailsOnDBErrorInTransaction(t, func(mock sqlmock.Sqlmock) {
+		mock.ExpectBegin()
+		mock.ExpectQuery(`SELECT .+ WHERE \(users\.ID = \?\)`).WithArgs(2).
+			WillReturnRows(sqlmock.NewRows([]string{"id"}).AddRow(2))
+		mock.ExpectQuery(regexp.QuoteMeta("SELECT groups.bFreeAccess FROM `groups` JOIN groups_ancestors ON groups_ancestors.idGroupChild = groups.ID WHERE (groups_ancestors.idGroupAncestor=?) AND (groups.ID = ?) LIMIT 1 FOR UPDATE")).
+			WithArgs(0, 1).WillReturnRows(sqlmock.NewRows([]string{"bFreeAccess"}).AddRow(true))
+		mock.ExpectExec("UPDATE `groups_groups` .+").WithArgs("requestRefused", 1).
+			WillReturnError(errors.New("some error"))
+		mock.ExpectRollback()
+	})
+}
+
+func TestService_updateGroup_ErrorOnUpdatingGroup(t *testing.T) {
+	assertUpdateGroupFailsOnDBErrorInTransaction(t, func(mock sqlmock.Sqlmock) {
+		mock.ExpectBegin()
+		mock.ExpectQuery(`SELECT .+ WHERE \(users\.ID = \?\)`).WithArgs(2).
+			WillReturnRows(sqlmock.NewRows([]string{"id"}).AddRow(2))
+		mock.ExpectQuery(regexp.QuoteMeta("SELECT groups.bFreeAccess FROM `groups` JOIN groups_ancestors ON groups_ancestors.idGroupChild = groups.ID WHERE (groups_ancestors.idGroupAncestor=?) AND (groups.ID = ?) LIMIT 1 FOR UPDATE")).
+			WithArgs(0, 1).WillReturnRows(sqlmock.NewRows([]string{"bFreeAccess"}).AddRow(false))
+		mock.ExpectExec("UPDATE `groups` .+").
+			WillReturnError(errors.New("some error"))
+		mock.ExpectRollback()
+	})
+}
+
+func assertUpdateGroupFailsOnDBErrorInTransaction(t *testing.T, setMockExpectationsFunc func(sqlmock.Sqlmock)) {
+	db, mock := database.NewDBMock()
+	defer func() { _ = db.Close() }()
+
+	setMockExpectationsFunc(mock)
+
+	base := service.Base{Store: database.NewDataStore(db), Config: nil}
+	srv := &Service{Base: base}
+	router := chi.NewRouter()
+	router.Post("/groups/{group_id}", service.AppHandler(srv.updateGroup).ServeHTTP)
+
+	monkey.Patch(auth.UserIDFromContext, func(context context.Context) int64 {
+		return 2
```

Comment:
Monkey patching should be seen as the last resort solution when there is no actual way to simulate a result in another way (so typically external libs or global state when was not avoidable). In this case, it is just a value put by a modular middleware, it should be quite easy to mock it properly with monkey patching. I can investigate how if you want. Or maybe I didn't understand the real need for that.

Reply:
I could not find another way of stubbing the user ID in the request's context. That's why `monkey.Patch` is used here.

Reply:
When I make it properly with a dummy middleware, I get a nil response... I don't understand. I'm on that for >30min, I give up.